### ISSUE TITLE
Mouse Capture Anomalies

### ISSUE THREAD

**Issue by RobertBColton** [enigma-dev/enigma-dev#1828]
This is a continuation of #1823 to address minor anomalies that still exist between ENIGMA and GM with regards to mouse behavior. What ENIGMA did in #1825 and #1824 was correct and an improvement. However, it's not exactly like GM since the capture can be prematurely released by press-and-releasing a different mouse button than the one that started the capture.

The problem here is that GM8.1 not only captures the mouse when a button is pressed inside the window, it seems to also record which button was pressed. From testing, I am able to show that it's possible to get GM8.1's mouse state stuck just like you can ENIGMA's. If we use the below draw code, we can actually press-and-move the left mouse button outside the window, which captures the mouse. Then we can press-and-hold the right mouse button and finally release the left button. GM8.1 at that point releases the mouse capture and if we release the right mouse button outside the window then, GM8.1 thinks it is still activated!

```gml
draw_text(0,0,string(mouse_check_button(mb_left)) + " " +
              string(mouse_check_button_pressed(mb_left)) + " " +
              string(mouse_check_button_released(mb_left)) + "#" +
              string(mouse_check_button(mb_right)) + " " +
              string(mouse_check_button_pressed(mb_right)) + " " +
              string(mouse_check_button_released(mb_right)) + "#" +
              string(keyboard_check(vk_space)) + " " +
              string(keyboard_check_pressed(vk_space)) + " " +
              string(keyboard_check_released(vk_space)));
```

This is **_NOT_** reproducible in GMSv1.4 and the issue seems to have been fixed!

I don't actually see this as an issue that we necessarily need to do anything about, but GMSv1.4 suggests YoYo did something about it. I think we could go years without anybody ever having a problem, but it is possible for us to easily adopt what GM8.1 does. However, since GM8.1 is also busted, I have my own proposal, which is that we just hold the mouse capture until **_ALL_** mouse buttons are released. We could also just read the mouse state directly.

**Comment by RobertBColton** [comment]
To further clarify, GM8.1 was actually even worse than ENIGMA after ENIGMA was initially fixed. GM8.1 totally ignored mouse capture for any mouse button that wasn't the left button. It behaved exactly like the official examples from Microsoft about how to use capture.

**Comment by RobertBColton** [comment]
Closing as resolved by #2039, we know for sure the mouse capture is working correctly everywhere now.
